appvm-scripts/usrbin/qubes-run-x11vnc-pre

```diff
@@ -0,0 +1,117 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+#
+
+# Source Qubes library.
+# shellcheck disable=SC1091
+. /usr/lib/qubes/init/functions
+
+array_min_max() {
+    read -r -a array <<< "$@"
+    max=${array[0]}
+    min=${array[0]}
+
+    for i in ${array[*]}; do
+      (( i > max )) && max=$i
+      (( i < min )) && min=$i
+    done
+    echo "$min"-"$max"
+}
+
+if qsvc guivm-vnc; then
+    # Common part from qubes-run-xorg to setup dummyqbs xorg.org
+    GUI_DOMID="$(qubesdb-read /qubes-gui-domain-xid 2>/dev/null)"
+    MEM_MIN="$(qubesdb-read /qubes-gui-videoram-min 2>/dev/null)"
+    MEM_OVERHEAD="$(qubesdb-read /qubes-gui-videoram-overhead 2>/dev/null)"
+    : "${MEM_MIN:=0}"
+    : "${MEM_OVERHEAD:=$((2560 * 1600 * 4 / 1024))}"
+
+    MEM=$((MEM + MEM_OVERHEAD))
+    if [ $MEM -lt $MEM_MIN ]; then
+        MEM=$MEM_MIN
+    fi
+
+    DUMMY_MAX_CLOCK=300
+    PREFERRED_HSYNC=50
+    DEPTH=24
+
+    HSYNC=()
+    VSYNC=()
+    MODELINES=()
+    # Set of common resolutions (VGA, VGA wide and HD)
+    RESOLUTIONS="320x240
+    640x480
+    800x600
+    1024x768
+    1152x864
+    1280x1024
+    1400x1050
+    1600x1200
+    2048x1536
+    800x480
+    1366x768
+    1440x900
+    1600x1024
+    1680x1050
+    1920x1200
+    2560x1600
+    1280x720
+    1920x1080"
+
+    for RES in ${RESOLUTIONS[*]};
+    do
+        # Defines every modeline like qubes.SetMonitorLayout
+        W=${RES%x*}
+        H=${RES#*x}
+        HTOTAL=$((W+3))
+        VTOTAL=$((H+3))
+        CLOCK=$((PREFERRED_HSYNC*HTOTAL/1000))
+        if [ $CLOCK -gt $DUMMY_MAX_CLOCK ]; then CLOCK=$DUMMY_MAX_CLOCK ; fi
+        MODELINE="$CLOCK $W $((W+1)) $((W+2)) $HTOTAL $H $((H+1)) $((H+2)) $VTOTAL"
+
+        HSYNC_START=$((CLOCK*1000/HTOTAL))
+        HSYNC_END=$((HSYNC_START+1))
+
+        VREFR_START=$((CLOCK*1000000/HTOTAL/VTOTAL))
+        VREFR_END=$((VREFR_START+1))
+
+        # Store computed h/v sync and Xorg Modeline entry
+        HSYNC+=("$HSYNC_START" "$HSYNC_END")
+        VSYNC+=("$VREFR_START" "$VREFR_END")
+        MODELINES+=("    Modeline \"QB$RES\" $MODELINE")
+    done
+
+    ALL_MODELINES="$(printf '%s\\n' "${MODELINES[@]}")"
+    MIN_MAX_HSYNC="$(array_min_max "${HSYNC[@]}")"
+    MIN_MAX_VSYNC="$(array_min_max "${VSYNC[@]}")"
+
+    # Render the template
+    sed -e s/%MEM%/$MEM/ \
+        -e s/%GUI_DOMID%/"$GUI_DOMID"/ \
+        -e s/%DEPTH%/$DEPTH/ \
+        -e s/%MODELINES%/"$ALL_MODELINES"/ \
+        -e s/%MIN_MAX_HSYNC%/"$MIN_MAX_HSYNC"/ \
+        -e s/%MIN_MAX_VSYNC%/"$MIN_MAX_VSYNC"/ \
+        < /etc/X11/xorg-qubes-x11vnc.conf.template > /etc/X11/xorg.conf
```

Comment:
It will be bad if it will override real xorg.conf that user has created. Or leave it even when vnc gui vm won't be used anymore (for example someone creates GUI VM as StandaloneVM, then initially uses it via VNC, but later tries with real GPU).
This script should make a backup of previous xorg.conf if it's there. And also cleanup after terminating lightdm (either remove xorg.conf if it wasn't there, or restore original one).